packages/blade/src/components/Typography/BaseText/BaseText.d.ts

```diff
@@ -0,0 +1,33 @@
+import type { ReactNode } from 'react';
+import type { ColorContrast } from '../../../tokens/theme/theme';
+import type { Theme } from '../../BladeProvider';
+
+type DotNotationStringToken<TokenType> = {
```

Comment:
this is to generate the `color` prop which supports values `action.text.primary.default`, `surface.text.muted.highContrast`, etc.

Reply:
We should put it in the jsdoc itself. 

```
/**
 * @template TokenType object
 * @description Tokenises objects to dot notation strings, eg: `a.b.c`
 */
```